### PR TITLE
Updated UN ILO topic generation script, added variable groupings

### PR DESCRIPTION
**Pull request by dwnoble**
* Updated UN ILO topic generation script 
* Added `dc/g/Custom_UN` variable grouping root to un staging environment
* Fixed some typos in undata who variable groupings
* Updated un staging ILO bt version
* added undata_ilo search index
* added undata_ilo embeddings and topic cache

### CODE REVIEW

File: deploy/nl/embeddings.yaml [hunk 1]
```diff
@@ -27,6 +27,10 @@ indexes:
     store: MEMORY
     embeddings: gs://datcom-nl-models/embeddings_undata_2024_03_20_11_01_12.ft_final_v20230717230459.all-MiniLM-L6-v2.csv
     model: ft-final-v20230717230459-all-MiniLM-L6-v2
+  undata_ilo_ft:
+    store: MEMORY
+    embeddings: gs://datcom-nl-models/embeddings_undata_ilo_2024_05_15_11_18_05.ft_final_v20230717230459.all-MiniLM-L6-v2.csv
+    model: ft-final-v20230717230459-all-MiniLM-L6-v2
```
Comment (pradh) [inline deploy/nl/embeddings.yaml:33]: nit: when you sync, you might also need to add this to https://github.com/datacommonsorg/website/blob/3a2854f075b4614baea488410469ba2feffd16e2/deploy/helm_charts/envs/unsdg.yaml#L45

File: server/lib/nl/explore/params.py [hunk 2]
```diff
@@ -49,6 +49,7 @@ class DCNames(str, Enum):
   SDG_DC = 'sdg'
   SDG_MINI_DC = 'sdgmini'
   UNDATA_DC = 'undata'
+  UNDATA_ILO_DC = 'undata_ilo'
```
Comment (pradh) [inline server/lib/nl/explore/params.py:52]: We should also add this to `SPECIAL_DC_LIST`

File: tools/sdg/topics/core_topics.py [hunk 3]
```diff
@@ -402,6 +424,21 @@ def main(_):
...
+    load_variables(vars_file=_UNDATA_ILO_VARIABLES_FILE,
+                   vars=vars,
+                   var_prefix='',
+                   topic_prefix='dc/topic/',
+                   svpg_prefix='dc/svpg/ILO')
+    write_non_country_vars(_UNDATA_ILO_NON_COUNTRY_VARS, vars)
+    nodes = generate([_UNDATA_ILO_ROOT],
+                     vars,
+                     REMOVE_SVG_PREFIX,
+                     skip_assert=True)
```
Comment (pradh) [inline tools/sdg/topics/core_topics.py:437]: Add a comment for why we skip_assert?  I suppose we want to resolve that (in the next iteration of variable groupings?)

File: tools/sdg/topics/core_topics.py [hunk 4]
```diff
@@ -48,20 +48,34 @@
...
+_UNDATA_ILO_ROOT = "dc/g/Custom_UN"
+_UNDATA_ILO_TOPIC_JSON = '../../../server/config/nl_page/undata_ilo_topic_cache.json'
+_UNDATA_ILO_NON_COUNTRY_VARS = '../../../server/config/nl_page/undata_ilo_non_country_vars.json'
+_UNDATA_ILO_MCF_PATH = os.path.join(_TMP_DIR, 'custom_topics_undata_ilo.mcf')
+_UNDATA_ILO_VARIABLES_FILE = 'undata_ilo_variable_grouping.csv'
+_UNDATA_ILO_SERIES_TOPICS_FILE = 'un_ilo_series_topics.csv'
+_UNDATA_ILO_NL_DESCRIPTIONS_FILE = os.path.join(
+    _TMP_DIR, 'undata_ilo_nl_descriptions.csv')
+
+REMOVE_SVG_PREFIX = "Custom_"
```
Comment (pradh) [inline tools/sdg/topics/core_topics.py:60]: Perhaps we could consider making this a flag like "--use_custom_dc_groups" or something that can then set "Custom_".  So when we run UNData post launch, we don't need it.

Deferring this to later is OK.

File: tools/sdg/topics/core_topics.py [hunk 5]
```diff
@@ -48,20 +48,34 @@
...
+
+API_ROOT = "https://staging.unsdg.datacommons.org"
```
Comment (pradh) [inline tools/sdg/topics/core_topics.py:62]: Maybe this too should not be hardcoded, but passed via flag?  Deferring to later is OK

File: tools/sdg/topics/core_topics.py [hunk 6]
```diff
@@ -48,20 +48,34 @@
...
+                     'dc/topic/sdg').replace('dc/g/', 'dc/topic/').replace(
+                         f"/{remove_svg_prefix}", "/")
 
 
-def download_svg_recursive(svgs: List[str], nodes: Dict[str, Dict],
-                           keep_vars: Set[str]):
+def download_svg_recursive(svgs: List[str],
+                           nodes: Dict[str, Dict],
+                           keep_vars: Set[str],
+                           remove_svg_prefix=""):
```
Comment (pradh) [inline tools/sdg/topics/core_topics.py:77]: I regret not wrapping this whole thing in a class :)

File: tools/sdg/topics/core_topics.py (hunk 3 above)
Comment (pradh) [inline tools/sdg/topics/core_topics.py:437]: Add a comment for why we skip_assert?  I suppose we want to resolve that (in the next iteration of variable groupings?)

File: deploy/nl/embeddings.yaml [hunk 7]
```diff
@@ -27,6 +27,10 @@ indexes:
     store: MEMORY
     embeddings: gs://datcom-nl-models/embeddings_undata_2024_03_20_11_01_12.ft_final_v20230717230459.all-MiniLM-L6-v2.csv
     model: ft-final-v20230717230459-all-MiniLM-L6-v2
+  undata_ilo_ft:
```
Comment (pradh) [inline deploy/nl/embeddings.yaml:30]: As validation, should we have an e2e test like https://github.com/datacommonsorg/website/blob/f7433df7f18fb0b306b7d04b101fe4701dfcf5eb/server/integration_tests/explore_test.py#L278-L281 ?

(And later when the data has been added to main DC, we can [do fulfill as well](https://github.com/datacommonsorg/website/blob/f7433df7f18fb0b306b7d04b101fe4701dfcf5eb/server/integration_tests/explore_test.py#L587-L590))

**Review by pradh — COMMENTED** [review]
Very cool!!

File: deploy/nl/embeddings.yaml (hunk 1 above)
Comment (dwnoble) [reply deploy/nl/embeddings.yaml:33]: Added!

File: server/lib/nl/explore/params.py (hunk 2 above)
Comment (dwnoble) [reply server/lib/nl/explore/params.py:52]: Added

File: tools/sdg/topics/core_topics.py (hunk 3 above)
Comment (dwnoble) [reply tools/sdg/topics/core_topics.py:437]: done

File: tools/sdg/topics/core_topics.py (hunk 4 above)
Comment (dwnoble) [reply tools/sdg/topics/core_topics.py:60]: Good callout- added a comment here for now, and can address this when moving this data to production

File: tools/sdg/topics/core_topics.py (hunk 5 above)
Comment (dwnoble) [reply tools/sdg/topics/core_topics.py:62]: +1 added a comment.

File: tools/sdg/topics/core_topics.py (hunk 6 above)
Comment (dwnoble) [reply tools/sdg/topics/core_topics.py:77]: After working with this script , maybe we can refactor it to a class and to take in a yaml config file for the various sdg/who/ilo configurations

**Comment by pradh** [comment]
I see your comments, but not the commit?

File: deploy/nl/embeddings.yaml (hunk 7 above)
Comment (dwnoble) [reply deploy/nl/embeddings.yaml:30]: Added

**Comment by dwnoble** [comment]
> I see your comments, but not the commit?

Commented a little too early- just pushed

**Review by pradh — APPROVED** [review]

**Pull request merged**